app/Http/Controllers/Api/AccessoriesController.php

```diff
@@ -141,18 +141,27 @@ public function checkedout($id, Request $request)
             return ['total' => 0, 'rows' => []];
         }
 
+        $offset = request('offset', 0);
+        $limit = request('limit', 50);
+
         $accessory->lastCheckoutArray = $accessory->lastCheckout->toArray();
         $accessory_users = $accessory->users;
-        
+        $total = $accessory_users->count();
+
+        if($total < $offset){
+            $offset = 0;
+        }
+
+        $accessory_users = $accessory->users()->skip($offset)->take($limit)->get();
+
         if ($request->filled('search')) {
             $accessory_users = $accessory->users()
                                 ->where('first_name', 'like', '%'.$request->input('search').'%')
                                 ->orWhere('last_name', 'like', '%'.$request->input('search').'%')
                                 ->get();
```

Comment:
This total variable is within the search constraint. If you’re not searching for something, would this variable be set?